SetupDataPkg/Tools/SettingSupport/UefiVariablesSupportLinuxLib.py

```diff
@@ -0,0 +1,139 @@
+# @file
+#
+# Python lib to support Reading and writing UEFI variables from Linux
+#
+# Copyright (c), Microsoft Corporation
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+
+import os
+import uuid
+import sys
+import struct
+
+from ctypes import (
+    create_string_buffer
+)
+
+EFI_VAR_MAX_BUFFER_SIZE = 1024 * 1024
+
+class UefiVariable(object):
+    ERROR_ENVVAR_NOT_FOUND = 0xcb
+
+    def __init__(self):
+        pass
+
+    #
+    # Function to get variable
+    # return a tuple of error code and variable data as string
+    #
+    def GetUefiVar(self, name, guid):
+        # success
+        err = 0
+        # the variable name is VariableName-Guid
+        path = '/sys/firmware/efi/efivars/' + name + '-%s' % guid
+
+        if not os.path.exists(path):
+            err = UefiVariable.ERROR_ENVVAR_NOT_FOUND
+            return (err, None)
+
+        efi_var = create_string_buffer(EFI_VAR_MAX_BUFFER_SIZE)
+        with open(path, 'rb') as fd:
+            efi_var = fd.read()
+
+        return (err, efi_var)
+
+    #
+    # Function to get all variable names
+    # return a tuple of error code and variable names byte array formatted as:
+    #
+    # typedef struct _VARIABLE_NAME {
+    #   ULONG NextEntryOffset;
+    #   GUID VendorGuid;
+    #   WCHAR Name[ANYSIZE_ARRAY];
+    # } VARIABLE_NAME, *PVARIABLE_NAME;
+    #
+    def GetUefiAllVarNames(self):
+        # success
+        status = 0
+
+        # implementation borrowed from https://github.com/awslabs/python-uefivars/blob/main/pyuefivars/efivarfs.py

```

Comment:
Since you referenced another repo with their own copyright and license, I think we need to maintain their copyright information, even though its a very minimal set of code that you reused.